code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,703 @@
+        '500':
+          $ref: '#/components/responses/500'
+        '503':
+          $ref: '#/components/responses/503'
+
+components:
+  securitySchemes:
+    oAuth2ClientCredentials:
+      type: oauth2
+      flows: 
+        clientCredentials: 
+          tokenUrl: '/oauth2/token'
+          scopes:
+            nbi-mgmt: Access to the APIs
+  
+  schemas:
+    AppId:
+      type: string
+      format: uuid
+      description: A globally unique identifier assosiated with the artifact. Originating OP generates this identifier when artifact is submitted over NBI.
+       
+    AppProviderId:
+      type: string
+      pattern: ^[A-Za-z][A-Za-z0-9_]{7,63}$
+      description: UserId of the app provider.  Identifier is relevant only in context of federation.
+    
+    AppProvisioning:
+      description: Attributes required to provision an application
+      type: object
+      properties:
+        artifactId:
+          $ref: '#/components/schemas/ArtifactId'
+        appVersion:
+          $ref: '#/components/schemas/AppVersion'
+        appProviderId:
+          $ref: '#/components/schemas/AppProviderId' 
+        edgeResource:
+          $ref: '#/components/schemas/EdgeResource'
+        regionId:
+          $ref: '#/components/schemas/RegionId'
+      required:
+        - artifactId
+    
+    AppVersion:
+      type: string
+      pattern:  ^(\d{1,2}\.)?(\d{1,2}\.)?(\d{1,2})$
+      description: Versioning info in the format major.minor.patch
+     
+    ArtifactId:
+      type: string
+      pattern: 6sa85a64-5117-3562-c3fd-2c913a66aba1
+      description: A globally unique identifier assosiated with the artifact. Originating OP generates this identifier when artifact is submitted over NBI.
+
+    ArtifactDetails:
+      properties:
+        name:
+          type: string
+          pattern: ^[A-Za-z][A-Za-z0-9_]{7,31}$
+          description: Name of the artifact. 
+        
+        version:
+          type: string
+          pattern: ^(\d{1,2}\.)?(\d{1,2}\.)?(\d{1,2})$
+          description: Artifact version information
+        
+        virtType:
+          type: string
+          enum:
+            - VM_TYPE
+            - CONTAINER_TYPE
+
+        fileName:
+          type: string
+          minLength:  8
+          maxLength: 32
+          description: Name of the file.  
+                    
+        fileFormat:
+          type: string
+          enum:
+            - WINZIP
+            - TAR
+            - TEXT
+            - TARGZ
+          description: Artifacts like Helm charts or Terraform scripts may need compressed format.
+                    
+        descriptorType:
+          type: string
+          enum:
+            - HELM
+            - TERRAFORM
+            - ANSIBLE
+            - SHELL      
+            - COMPONENTSPEC
+          description: Type of descriptor present in the artifact.  App provider can either define either a HELM chart or a Terrform script or container spec.
```

Comment:
This would also mean that component spec goes separately. Which also makes sense, becuase the same VM could be deployed using different flavours based on HW or SLAs. That is, VM `a` could be deployed with `2 CPUs, 2 GB DRAM` but also with `8 CPUs, 8 GB DRAM` if it's required.